### ISSUE TITLE
Heartbeats resources don't come up after using the Eventing operator

### ISSUE THREAD

**Issue by JamesMcMahon** [knative/operator#847]
**Describe the bug**
Following the instructions at https://knative.dev/docs/admin/install/knative-with-operators/#installing-the-knative-eventing-component.

I end up with the following broken resources, (non-broken omitted for clarity)

```
NAME                                                           READY   STATUS             RESTARTS   AGE
pod/heartbeats-source-deployment-64dff6896d-lqslh              0/1     CrashLoopBackOff   255        21h

NAME                                           READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/heartbeats-source-deployment   0/1     1            0           21h

NAME                                                      DESIRED   CURRENT   READY   AGE
replicaset.apps/heartbeats-source-deployment-64dff6896d   1         1         0       21h

NAME                                                    SINK   AGE   READY   REASON
containersource.sources.knative.dev/heartbeats-source          21h   False   MinimumReplicasUnavailable

NAME                                                            SINK   AGE   READY   REASON
sinkbinding.sources.knative.dev/heartbeats-source-sinkbinding          21h   False   NoURI
```

Logs for the broken pod,

```
2021/10/29 17:03:08 Failed to read tracing config, using the no-op default: empty json tracing config
2021/10/29 17:03:08 failed to create client: http target option was empty string
```

Describing the sinkbinding source shows the following status,

```yaml
Status:
  Conditions:
    Last Transition Time:  2021-10-28T19:31:47Z
    Message:               URI could not be extracted from destination
    Reason:                NoURI
    Status:                False
    Type:                  Ready
    Last Transition Time:  2021-10-28T19:31:47Z
    Status:                Unknown
    Type:                  SinkProvided
```

**Expected behavior**
The resources should be brought up by the operator. If additional config or setup is needed it should be documented.

**To Reproduce**
Kubectl applying the following
```
apiVersion: v1
kind: Namespace
metadata:
  name: knative-eventing
---
apiVersion: operator.knative.dev/v1alpha1
kind: KnativeEventing
metadata:
  name: knative-eventing
  namespace: knative-eventing
```

**Knative release version**
```
NAME               VERSION   READY   REASON
knative-eventing   0.26.0    True
````

**Additional context**
Add any other context about the problem here such as proposed priority

**Comment by xtreme-sameer-vohra** [comment]
Hey @JamesMcMahon 
Can you share the output of
```
kubectl -n knative-eventing get all

kubectl logs pod/heartbeats-source-deployment-64dff6896d-lqslh
```

From the info you shared, it seems like the `containersource.sources.knative.dev/heartbeats-source` is not in `ready` state and its likely due to `pod/heartbeats-source-deployment-64dff6896d-lqslh` being in a `CrashLoopBackOff`.

**Comment by JamesMcMahon** [comment]
`kubectl -n knative-eventing get all`

Output:
```
NAME                                                           READY   STATUS             RESTARTS   AGE
pod/eventing-controller-7c9b5b58f7-x9xbw                       1/1     Running            0          4d18h
pod/eventing-webhook-9cd57d59d-6wg6w                           1/1     Running            0          4d18h
pod/heartbeats-source-deployment-64dff6896d-lqslh              0/1     CrashLoopBackOff   1349       4d18h
pod/imc-controller-6f8756c89b-dgkp8                            1/1     Running            0          4d18h
pod/imc-dispatcher-66b9954cb5-cpt4b                            1/1     Running            0          4d18h
pod/mt-broker-controller-758f4dcbb5-v42m6                      1/1     Running            0          4d18h
pod/mt-broker-filter-86f6c597bf-4r5dn                          1/1     Running            0          4d18h
pod/mt-broker-ingress-59d65668c-k8tmh                          1/1     Running            0          4d18h
pod/rabbitmq-controller-manager-678d699db9-hnt8x               1/1     Running            0          4d18h
pod/rabbitmq-webhook-7464c9d64b-t8ncw                          1/1     Running            0          4d18h
pod/storage-version-migration-eventing-eventing-0.26.0-kn9nl   0/1     Completed          0          4d18h
pod/sugar-controller-6f55dcffb5-kv2xc                          1/1     Running            0          4d18h

NAME                              TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                     AGE
service/broker-filter             ClusterIP   10.120.200.254   <none>        80/TCP,9092/TCP             4d18h
service/broker-ingress            ClusterIP   10.120.200.64    <none>        80/TCP,9092/TCP             4d18h
service/eventing-webhook          ClusterIP   10.120.200.132   <none>        443/TCP                     4d18h
service/imc-dispatcher            ClusterIP   10.120.200.11    <none>        80/TCP,9090/TCP             4d18h
service/inmemorychannel-webhook   ClusterIP   10.120.200.133   <none>        443/TCP,9090/TCP,8008/TCP   4d18h
service/rabbitmq-controller       ClusterIP   10.120.200.98    <none>        443/TCP                     4d18h
service/rabbitmq-webhook          ClusterIP   10.120.200.189   <none>        443/TCP                     4d18h

NAME                                           READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/eventing-controller            1/1     1            1           4d18h
deployment.apps/eventing-webhook               1/1     1            1           4d18h
deployment.apps/heartbeats-source-deployment   0/1     1            0           4d18h
deployment.apps/imc-controller                 1/1     1            1           4d18h
deployment.apps/imc-dispatcher                 1/1     1            1           4d18h
deployment.apps/mt-broker-controller           1/1     1            1           4d18h
deployment.apps/mt-broker-filter               1/1     1            1           4d18h
deployment.apps/mt-broker-ingress              1/1     1            1           4d18h
deployment.apps/pingsource-mt-adapter          0/0     0            0           4d18h
deployment.apps/rabbitmq-controller-manager    1/1     1            1           4d18h
deployment.apps/rabbitmq-webhook               1/1     1            1           4d18h
deployment.apps/sugar-controller               1/1     1            1           4d18h

NAME                                                      DESIRED   CURRENT   READY   AGE
replicaset.apps/eventing-controller-7c9b5b58f7            1         1         1       4d18h
replicaset.apps/eventing-webhook-9cd57d59d                1         1         1       4d18h
replicaset.apps/heartbeats-source-deployment-64dff6896d   1         1         0       4d18h
replicaset.apps/imc-controller-6f8756c89b                 1         1         1       4d18h
replicaset.apps/imc-dispatcher-66b9954cb5                 1         1         1       4d18h
replicaset.apps/mt-broker-controller-758f4dcbb5           1         1         1       4d18h
replicaset.apps/mt-broker-filter-86f6c597bf               1         1         1       4d18h
replicaset.apps/mt-broker-ingress-59d65668c               1         1         1       4d18h
replicaset.apps/pingsource-mt-adapter-b5b8fbd96           0         0         0       4d18h
replicaset.apps/rabbitmq-controller-manager-678d699db9    1         1         1       4d18h
replicaset.apps/rabbitmq-webhook-7464c9d64b               1         1         1       4d18h
replicaset.apps/sugar-controller-6f55dcffb5               1         1         1       4d18h

NAME                                                     REFERENCE                      TARGETS   MINPODS   MAXPODS   REPLICAS   AGE
horizontalpodautoscaler.autoscaling/broker-filter-hpa    Deployment/mt-broker-filter    1%/70%    1         10        1          4d18h
horizontalpodautoscaler.autoscaling/broker-ingress-hpa   Deployment/mt-broker-ingress   1%/70%    1         10        1          4d18h
horizontalpodautoscaler.autoscaling/eventing-webhook     Deployment/eventing-webhook    9%/100%   1         5         1          4d18h

NAME                                                           COMPLETIONS   DURATION   AGE
job.batch/storage-version-migration-eventing-eventing-0.26.0   1/1           6s         4d18h

NAME                                                    SINK   AGE     READY   REASON
containersource.sources.knative.dev/heartbeats-source          4d18h   False   MinimumReplicasUnavailable

NAME                                                            SINK   AGE     READY   REASON
sinkbinding.sources.knative.dev/heartbeats-source-sinkbinding          4d18h   False   NoURI
```

```
kubectl logs pod/heartbeats-source-deployment-64dff6896d-lqslh
```

Output:

```
2021/11/02 14:11:23 Failed to read tracing config, using the no-op default: empty json tracing config
2021/11/02 14:11:23 failed to create client: http target option was empty string
```

Will make an attempt to update with environment variables and see what happens.

**Comment by xtreme-sameer-vohra** [comment]
Thanks @JamesMcMahon 
The eventing services appear to be running, so this isn't an issue with [the operator](https://github.com/knative/operator) and this issue should be closed.
However, there is a problem (you can confirm if adding the environment variable solves it) and then we can file an issue with `kn` or `ContainerSource` 

cc @lionelvillard if you can help direct if this is an `kn` or `ContainerSource` issue

**Comment by JamesMcMahon** [comment]
If the broker connection needs additional environmental configuration should that be part of the docs around installing the operator?

**Comment by xtreme-sameer-vohra** [comment]
> If the broker connection needs additional environmental configuration should that be part of the docs around installing the operator?

The changes by applying
```
kn source container update <NAME OF THE SOURCE> -e POD_NAMESPACE=<NAMESPACE OF THE SOURCE> -e POD_NAME=<NAME OF THE SOURCE>
```
would be applied to the [ContainerSource](https://knative.dev/docs/eventing/custom-event-source/containersource/). If you can confirm that it resolves the problem, it confirms that the issue is either with the `kn` CLI or `ContainerSource` as I was able to reproduce it without using the operator to install Knative.

**Comment by JamesMcMahon** [comment]
I was trying to get my namespace in a clean room state to try your suggested solution today and I saw that version 1.0.0 just dropped so I figured I would try that.

The entire namespace now comes up correctly with no additional configuration and there are no resources referring to `heart-beats` anymore.

So I think this may have just been fixed by the latest release.

I am still confused about how I would go about connecting a broker and source to Knative eventing but that feels like that is another thread of conversation.

Closing this out as it looks like no longer a valid issue.
